### ISSUE TITLE
RecordBatch requires all its arrays to have an equal number of rows when running pipeline in streaming mode

### ISSUE THREAD

**Issue by EpicUsaMan** [pola-rs/polars#18599]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python

```

Can't guarantee any reproducible example, because it happens only on really large frames

### Log output

```shell
thread '<unnamed>' panicked at /home/runner/work/polars/polars/crates/polars-arrow/src/record_batch.rs:22:31:
called `Result::unwrap()` on an `Err` value: ComputeError(ErrString("RecordBatch requires all its arrays to have an equal number of rows"))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'polars-5' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-51' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-2' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-8' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-54' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-44' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-15' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-50' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-45' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-7' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-40' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-48' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-26' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-49' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-16' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-36' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-13' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-12' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-4' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
thread 'polars-18' panicked at crates/polars-pipe/src/executors/sinks/io.rs:271:49:
called `Result::unwrap()` on an `Err` value: "SendError(..)"
```

### Issue description

This error happen ONLY when running query with streaming=True and starts to appear after upgrade to polars 1.6.0

my flow is:
scan_parquet -> filter -> sort -> cum_sum / ewm_mean -> group_by

(Happens almost immediately after running pipeline, so looks like falling during scan/sort)

### Expected behavior

Streaming to work exactly the same as regular pipeline

### Installed versions

<details>

```
--------Version info---------
Polars:              1.6.0
Index type:          UInt32
Platform:            Linux-6.8.0-41-generic-x86_64-with-glibc2.39
Python:              3.12.3 (main, Jul 31 2024, 17:43:48) [GCC 13.2.0]

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               <not installed>
cloudpickle          3.0.0
connectorx           <not installed>
deltalake            <not installed>
fastexcel            <not installed>
fsspec               2024.6.1
gevent               24.2.1
great_tables         <not installed>
matplotlib           <not installed>
nest_asyncio         1.6.0
numpy                1.26.4
openpyxl             <not installed>
pandas               2.2.2
pyarrow              16.1.0
pydantic             <not installed>
pyiceberg            <not installed>
sqlalchemy           2.0.31
torch                2.3.1+rocm6.0
xlsx2csv             <not installed>
xlsxwriter           <not installed>
```

</details>

**Comment by ritchie46** [comment]
Without a reproducable example this isn't something we can fix sadly. :/ 

Though we are making a whole new engine and removing the old streaming one, so it might get fixed implicitly.

**Comment by EpicUsaMan** [comment]
> Without a reproducable example this isn't something we can fix sadly. :/
> 
> Though we are making a whole new engine and removing the old streaming one, so it might get fixed implicitly.

It's actually just somehow in streaming we are losing error, which I described here, but still can't prepare minimal example
It produce a bug only on very large pipelines during streaming
Removing .over() by columns with only 1 value fixes the problem
https://github.com/pola-rs/polars/issues/18600

**Comment by nik-sm** [comment]
I'm encountering this issue but on latest version (1.16.0).

The issue is reproducible in my codebase, but I'm not able to export the dataframe and make a small standalone repro. Here's what I've observed, and please let me know if there are more details I can give, or what other bits of state I should save along with the actual dataframe in order to make a reproducible example.


The dataframe looks like this:
```
ipdb> df.head()
shape: (5, 4)
┌─────────────────────┬──────────┬────────────────────────┬─────────────┐
│ timestamp           ┆ solution ┆ is_within_fluidics_roi ┆ solution_id │
│ ---                 ┆ ---      ┆ ---                    ┆ ---         │
│ datetime[ms]        ┆ str      ┆ bool                   ┆ i32         │
╞═════════════════════╪══════════╪════════════════════════╪═════════════╡
│ 2021-11-08 09:40:26 ┆ CAL      ┆ false                  ┆ 0           │
│ 2021-11-08 09:43:26 ┆ CAL      ┆ true                   ┆ 1           │
│ 2021-11-08 09:44:26 ┆ CAL      ┆ false                  ┆ 1           │
│ 2021-11-08 09:44:35 ┆ CAL      ┆ false                  ┆ 1           │
│ 2021-11-08 09:47:36 ┆ CAL      ┆ true                   ┆ 2           │
└─────────────────────┴──────────┴────────────────────────┴─────────────┘
ipdb> df.schema
Schema({'timestamp': Datetime(time_unit='ms', time_zone=None), 'solution': String, 'is_within_fluidics_roi': Boolean, 'solution_id': Int32})
ipdb> df.shape
(194, 4)
```

The failing code is:
```python
df.write_avro("foobar.avro")

thread '<unnamed>' panicked at /Users/runner/work/polars/polars/crates/polars-arrow/src/record_batch.rs:25:39:
called `Result::unwrap()` on an `Err` value: ComputeError(ErrString("RecordBatch requires all its arrays to have an equal number of rows"))
*** pyo3_runtime.PanicException: called `Result::unwrap()` on an `Err` value: ComputeError(ErrString("RecordBatch requires all its arrays to have an equal number of rows"))

```

I thought I might make a repro by just saving that dataframe in another format, loading it from a fresh python REPL and then trying `.write_avro()`. However, this does not work.
I tried to export the failing dataframe with `df.write_csv()`, `df.write_parquet()`, and as a pickled object. But in all those cases, when loading the "bad" dataframe from a fresh REPL, I find that I can do `freshly_loaded_df.write_avro("foobar.avro")` with no error.

So it must be that there is an additional piece of state/context in my failing use case that causes the issue.

Note that exporting the same dataframe into these other formats (CSV, Parquet) doesn't cause an error, which makes me pretty sure that the issue is within `.write_avro()` part


Here's my version info: 

<details>
<summary> versions </summary>

```
ipdb> pl.show_versions()
--------Version info---------
Polars:              1.16.0
Index type:          UInt32
Platform:            macOS-15.1.1-arm64-arm-64bit
Python:              3.12.6 (main, Sep 27 2024, 17:59:45) [Clang 15.0.0 (clang-1500.3.9.4)]
LTS CPU:             False

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               <not installed>
boto3                <not installed>
cloudpickle          <not installed>
connectorx           <not installed>
deltalake            <not installed>
fastexcel            0.11.6
fsspec               2024.9.0
gevent               <not installed>
google.auth          <not installed>
great_tables         <not installed>
matplotlib           <not installed>
nest_asyncio         1.6.0
numpy                2.1.2
openpyxl             <not installed>
pandas               <not installed>
pyarrow              17.0.0
pydantic             2.9.2
pyiceberg            <not installed>
sqlalchemy           <not installed>
torch                2.5.1
xlsx2csv             <not installed>
xlsxwriter           <not installed>
```

</details>

**Comment by YVin3D** [comment]
Hi, i solved this issue when it popped up for me.

## TLDR How to fix

You have a duplicate column that polars doesn’t know is duplicate until write time. Not sure why

To find out the problematic column, run this:

`df = df.select(df.columns)`

This will give you an error with the problematic column name.

## My Issue

Bug was that i had a df0 with a column name, let’s say colA_right 

Then i ran join_asof on it with another, df1, that had another column called colA. The merge adds the suffix to the newly merged df also called colA_right, but for some reason, one of my filters or operations causes this to not throw an error.

Idk if this will help close the ticket but it may help someone else struggling with this. Please someone provide a repro

**Comment by cmdlineluser** [comment]
@YVin3D That sounds a bit different and more like an optimizer issue e.g. 

- https://github.com/pola-rs/polars/issues/20831

**Comment by coastalwhite** [comment]
I am going to close this issue as there is no reproducer and there is a completely different streaming engein now.
